sktalk/corpus/parsing/parser.py

```diff
@@ -20,9 +19,17 @@ def metadata(self):
             return self._metadata | metadata
         raise ValueError("Duplicate key in the metadata")
 
+    @property
+    def utterances(self):
+        self._utterances = self._extract_utterances()
+        return self._utterances
```

Comment:
Perhaps this is not a real concern at this moment, but if there is a sub-class that implements a non-trivial `_extract_utterances()` method, that method is executed every time that the `utterances` property is called. However, properties are typically expected to be computationally trivial. Perhaps check if the utterances have been extracted before.

Anyway, this requires a change in `__init__()` above too because you need to differ between the cases where `self._utterances` is empty (`[]`) or not yet extracted (`None`) (see suggestion above).

```suggestion
        if self._utterances is None:
            self._utterances = self._extract_utterances()
        return self._utterances
```

Reply:
Given the nature of the class, this is not a concern, and perhaps even a feature: when calling ChaFile.parse(), indeed the intended behavior is that the file be parsed, which calls both _extract_metadata and _extract_utterances.

I'm actually more concerned with the properties `metadata` and `utterances` being available at all; that said, this PR rewrites it such that we never really ask a user to instantiate a class of ChaFile (or any other input format), so they will not generate it nor access those properties.